pkg/storage/etcd3/watcher.go

```diff
@@ -276,6 +283,12 @@ func parseError(err error) *watch.Event {
 }
 
 func (wc *watchChan) sendError(err error) {
+	// Context.canceled is an expected behavior.
+	// We should just stop all goroutines in watchChan without returning error.
+	// TODO: etcd client should return context.Canceled instead of grpc specific error.
+	if grpc.Code(err) == codes.Canceled || err == context.Canceled {
```

Comment:
The comment explains the problem (which I understand) not why you are using grpc.

>
Currently etcd client returns grpc error. It shouldn't though. We will fix it later by return "context.Canceled"

ok - that makes sense.